src/trie.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+from collections import deque, OrderedDict
+
+
+class Trie(object):
+    def __init__(self, iterable=None):
+        self.root = OrderedDict()
+        if iterable:
+            for i in iterable:
+                self.insert(i)
+
+    def insert(self, token):
+        start = self.root
+        for letter in token:
+            start = start.setdefault(letter, OrderedDict())
+        start['#'] = '#'
+
+    def contains(self, token):
+        """Verify if a token is in the trie."""
+        start = self.root
+        counter = 0
+        while counter < len(token):
+            try:
+                start = start[token[counter]]
+            except KeyError:
+                return False
+            counter += 1
+            try:
+                return start['#'] == '#'
+            except KeyError:
+                pass
```

Comment:
I spent some more time analyzing your algorithm. It has a problem where it incorrectly returns `True` when you check contains() on a stub of an inserted item.

e.g. when you insert "Python", contains("Py") should return False, but you're getting True.

I identified this line as the issue. Consider it very closely. What scenario is leading to this line being hit? What should you do instead?